MetaMask/metamask-extension
# fix/#20498

## **Description**
This PR is fixes#20498 . Which is to Replace the old Menu Component with new Popover Component.

_1. What is the reason for the change?_
Ans: @deprecated The `<Menu />` component has been deprecated in favor of the new `<Popover>` component from the component-library.

_2. What is the improvement/solution?_
Ans: In this PR, I've changed just `GlobalMenu` component.

## **Manual testing steps**

_1. Step1:_
👉 Open `metamask` extension, Enter your Password to login

_2. Step2:_
👉 Click on `Three Dots` on the `Top Right` side of the header. to see the changes


## **Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change._

### **Before with `Menu Component`**

![Menu_Component](https://github.com/MetaMask/metamask-extension/assets/91899494/020f709f-b7c9-4e2c-94b0-4f80a79edfbd)


### **After with `Popover Component`**

![Popover_Component](https://github.com/MetaMask/metamask-extension/assets/91899494/80eb7c86-35c6-4c9f-b9e9-dd7275f3c544)


## **Related issues**

One Issue that I want to tell you is that, You have to close `Popover Component` to click on three dots again.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
